test/unit/checker_test.rb

```diff
@@ -0,0 +1,17 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+module Packwerk
+  class CheckerTest < Minitest::Test
+    include FactoryHelper
+
+    test "#find is correctly able to find the right checker" do
+      # Load in the checker so its registered
+      require "packwerk/reference_checking/checkers/dependency_checker"
```

Comment:
I assume https://github.com/Shopify/packwerk/blob/72aa3cf9dd1c4035356e5474e115583bada071da/lib/packwerk/run_context.rb#L12 is implicitly loading this in non-test runs. Should we just add a require to the main entrypoint like we did with the offense formatter?

Reply:
Yep, makes a lot of sense. This was also able to remove the unused `DEFAULT_CHECKERS` param which was just implicitly used to register the constant.